tanglegram/tangle.py

```diff
@@ -720,7 +620,6 @@ def shuffle_dendogram(link, copy=True):
 
 def leaf_order(link, labels=None, as_dict=True):
```

Comment:
I still find the entanglement acting so weird. I might know the reason. The problem comes from the leaf_order function.

- `leafs_ix = sclust.hierarchy.leaves_list(link)` returns a list of indices of objects as they appear in the dendrogram (these indices are corresponding with the indices in "labels")
- ` if as_dict:
        if not isinstance(labels, type(None)):
            return dict(zip(labels, leafs_ix))`
matches the labels in "labels" with indices in sclust.hierarchy.leaves_list(link)
- However,  the orders of objects in "labels" and in the x-axis of the dendrogram are different, so the matching is wrong.

I can give you an example via email.